### ISSUE TITLE
PoxyShape.primPath and usd xforms in vp2RenderDelegate

### ISSUE THREAD

**Issue by jufrantz** [Autodesk/maya-usd#1535]
**Describe the bug**

Since maya-usd 0.9, vp2RenderDelegate takes the `primPath` attribute into account and that's a great addition ! But unfortunately we can't get the same desired result as with maya-usd 0.8 and GLBatchRenderer. Now setting primPath isolates the branch in the viewport, preserving its usd world xform. This is an issue for some of our workflows that need to "reference" in maya an usd branch rendered at proxyShape's xform.

**Steps to reproduce**
Launch this script in maya with 
- maya-usd-0.8.0 and MAYAUSD_DISABLE_VP2_RENDER_DELEGATE=1
- and maya-usd-0.9.0

```python
from pxr import Usd, UsdGeom
from maya import cmds as mc

# Create an usd hierarchy with some xformOps
stage = Usd.Stage.CreateInMemory()
xform = UsdGeom.Xform.Define(stage, '/xform')
xform.AddTranslateOp().Set((0.0, 10.0, 0.0))

cube1 = UsdGeom.Cube.Define(stage, '/xform/cube1')
cube1.AddTranslateOp().Set((-10.0, 0.0, 0.0))

cube2 = UsdGeom.Cube.Define(stage, '/xform/cube2')
cube2.AddTranslateOp().Set((10.0, 0.0, 0.0))

# Create a proxyShape referencing xformed mesh, results in the viewport differ in maya-usd 0.8 and 0.9
mc.loadPlugin('mayaUsdPlugin', quiet=True)
mc.createNode('mayaUsdProxyShape')
mc.setAttr('.filePath', stage.GetRootLayer().identifier, type='string')
mc.setAttr('.primPath', cube2.GetPath(), type='string')
```

**Expected behavior**
Same result in maya-usd 0.9.0 as we had with 0.8.0, `/xform/cube2` rendered at world origin.

**Specs (if applicable):**
 - Linux CentOS 7.4
 - Maya 2022
 - Maya USD v0.9.0
 - Pixar USD v21.05

**Additional context**
We are not using directly mayaUsdProxyShape as in the repro script but a custom proxyShape type inheriting from ProxyShapeBase. A solution involving changes to our custom shape could be applicable.

Best regards,
Julien Frantz

**Comment by jufrantz** [comment]
Hello,
Are there any news about this issue, would it need more details from our side ?
I noticed that it looks very similar to #1569 in backlog.

**Comment by santosg87** [comment]
hi @jufrantz 

I am able to reproduce the issue you mention.

interestingly enough, this seems to only happen while the user is in the anonymous layer, once that gets written to disk, the data is as expected. I save both a usd file from 0.8.0 and 0.9.0 versions, and they are identical after writing to disk. I am investigating a bit deeper on what causes the prim path to be set on the proxyShape node.

**Comment by jufrantz** [comment]
Hello @santosg87,

Setting the primPath in the repro steps is intentional. I tried with a layer on disk and I get same (unwanted) behaviour in maya-usd >= 0.9.0. 

1) Create an usd hierarchy with some xformOps, save layer to disk
```python
from pxr import Usd, UsdGeom, Sdf
import os.path

destFile = os.path.expandvars('$TMPDIR/cubesWithXforms.usd')
destLayer = Sdf.Layer.CreateNew(destFile)
stage = Usd.Stage.Open(destLayer)

xform = UsdGeom.Xform.Define(stage, '/xform')
xform.AddTranslateOp().Set((0.0, 10.0, 0.0))

cube1 = UsdGeom.Cube.Define(stage, '/xform/cube1')
cube1.AddTranslateOp().Set((-10.0, 0.0, 0.0))

cube2 = UsdGeom.Cube.Define(stage, '/xform/cube2')
cube2.AddTranslateOp().Set((10.0, 0.0, 0.0))

destLayer.Save()
```

2) Create a proxyShape and set its primPath attrib to render only cube2

```python
from maya import cmds as mc
import os.path
mc.loadPlugin('mayaUsdPlugin', quiet=True)

sourceFile = os.path.expandvars('$TMPDIR/cubesWithXforms.usd')

mc.createNode('mayaUsdProxyShape')
mc.setAttr('.filePath', sourceFile, type='string')
mc.setAttr('.primPath', '/xform/cube2', type='string')
```

**Comment by wtelford** [comment]
@jufrantz I don't believe I understand the ask here. The only difference between 0.8.0 and 0.9.0 should be what prims are rendered, not where the prims' are in worldspace.

Given the transform hierarchy:
- xform
  - cube1
  -  cube2
...and a primpath set to '/xform/cube2', you would get the following results:

In 0.8.0 both cubes are visible in their correct worldspace.
In 0.9.0 only /xform/cube2 is visible, but it is still in it's correct worldspace.

What is your expectation here?

**Comment by jufrantz** [comment]
Hi @wtelford,

Thank you for your answer and sorry if my description was not clear.

The difference is between `0.9.0` and `0.8.0 with env MAYAUSD_DISABLE_VP2_RENDER_DELEGATE=1`.

I compared against 0.9.0 vp2RenderDelegate because we were moving from pxrUsdMayaGL to vp2RenderDelegate with this version.

I would expect to be able to render only "/xform/cube2" but in its usd localspace.

Julien Frantz.

**Comment by santosg87** [comment]
@jufrantz would you be able to make a quick screen capture of the behaviour? I am trying to understand the issue, but I am a bit confused trying to wrap my head around it. :)
